### PR TITLE
Performance Tests [for comparison]

### PR DESCRIPTION
**Pull request by johannes-wolf**
Adds some tests with a bigger model to run under valgrind.

DO NOT MERGE. There is a rebased version of this branch for #23: #30

### CODE REVIEW

File: test/performance.cpp [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+    for (const auto& vv : res) {
+        if (!vals.empty())
+            vals.push_back('|');
+        vals += vv.toString();
+    }
+    return vals;
+}
+
+TEST_CASE("Small model performance queries", "[perf.big-model-benchmark]") {
+    const auto model = generate_model(25);
```
Comment (josephbirkner) [inline test/performance.cpp:125]: Is there a reason not to do the benchmark with the bigger model?

Comment (johannes-wolf) [reply test/performance.cpp:125]: Have to somehow exclude them from running under valgrind, as it becomes too slow.

Comment (johannes-wolf) [reply test/performance.cpp:125]: Fixed in the rebased version.

**Comment by johannes-wolf** [comment]
```
25: benchmark name                       samples       iterations    estimated
25:                                      mean          low mean      high mean
25:                                      std dev       low std dev   high std dev
25: -------------------------------------------------------------------------------
25: Query typeof Recursive                         100             1     3.44935 s 
25:                                         34.4949 ms    34.4846 ms     34.508 ms 
25:                                           59.02 us    47.1425 us    75.8362 us 
25:                                                                                
25: Query field id Recursive                       100             1     2.09223 s 
25:                                         20.4813 ms    20.4315 ms     20.537 ms 
25:                                         268.792 us     245.06 us    286.361 us 
25:                                                                                
25: Query field id                                 100             1    32.6139 ms 
25:                                          324.59 us    324.031 us    325.933 us 
25:                                         4.17649 us    2.19362 us    8.46351 us 
```

**Pull request closed without merging**